tests/MongoDB.Driver.Core.Tests/Core/Authentication/ScramSha1AuthenticatorTests.cs

```diff
@@ -155,26 +156,41 @@ public void Constructor_should_throw_an_ArgumentNullException_when_credential_is
                    ok: 1}"));
 
             var connection = new MockConnection(__serverId);
-            connection.EnqueueReplyMessage(saslStartReply);
+            var isMasterResult = (BsonDocument)__description.IsMasterResult.Wrapped.Clone();
+            if (useSpeculativeAuthenticate)
+            {
+                isMasterResult.Add("speculativeAuthenticate", saslStartReply.Documents[0].ToBsonDocument());
+            }
+
+            /* set buildInfoResult to 3.4 to force authenticator to use Query Message Wire Protocol because MockConnection
+             * does not support OP_MSG */
+            connection.Description = new ConnectionDescription(
+                __description.ConnectionId, new IsMasterResult(isMasterResult), new BuildInfoResult(new BsonDocument("version","3.4")));
```

Comment:
I admit this is slightly hacky, so an argument could be made for adding OP_MSG support to `MockConnection`. However, I would argue in this test, we're not testing HOW the messages are sent over the wire (and we know that OP_MSG works from other tests); rather in this test, the goal is to ensure that the **contents**  of the messages are correct so that authentication succeeds.

Reply:
Also we were already doing this for the other SCRAM-SHA-256 authentication tests, so there's also precedent :)